File: main/lib/models/grupo.dart
import 'dart:convert';

Grupo grupoFromJson(String str) => Grupo.fromJson(json.decode(str));

String grupoToJson(Grupo data) => json.encode(data.toJson());

class Grupo {
  Grupo({
    this.codigo,
    this.nombre,
    this.avatar,
    this.descripcion,
    this.usuarioCrea,
    this.privateKey,
    this.publicKey,
    this.fecha,
  });

  String? nombre;
  String? descripcion;
  String? avatar;
  String? fecha;
  String? usuarioCrea;
  String? codigo;
  String? privateKey;
  String? publicKey;

  factory Grupo.fromJson(Map<String, dynamic> json) => Grupo(
        codigo: json['codigo'],
        nombre: json["nombre"],
        descripcion: json["descripcion"],
        avatar: json['avatar'],
        usuarioCrea: json['usuarioCrea'],
        fecha: json['fecha'],
        privateKey: json['privateKey'],
        publicKey: json['publicKey'],
      );

  Map<String, dynamic> toJson() => {
        "codigo": codigo,
        "nombre": nombre,
        "descripcion": descripcion,
        "avatar": avatar,
        "usuarioCrea": usuarioCrea,
        "fecha": fecha,
        "privateKey": privateKey,
        "publicKey": publicKey,
      };
}

File: locksy-backend-main/models/grupo.js
const { Schema, model } = require("mongoose");
var Usuario = model("Usuario");

const GrupoSchema = Schema({
  codigo: {
    type: String,
    required: true,
  },
  nombre: {
    type: String,
    required: true,
  },
  avatar: {
    type: String,
  },
  descripcion: {
    type: String,
  },
  usuarioCrea: {
    type: Schema.ObjectId,
    ref: "Usuario",
  },
  publicKey: {
    type: String,
  },
  privateKey: {
    type: String,
  },
  fecha: {
    type: String,
  },
  disappearMessageTime: {
    type: String,
    enum: ["0m", "1m", "5m", "15m", "12h", "24h", "48h", "7d", "14d", "21d"],
  },
  disappearMessageSetAt: {
    type: Date,
  },
  disappearedCheck: {
    type: Boolean,
    default: false,
  },
});

GrupoSchema.method("toJSON", function () {
  const { _id, ...object } = this.toObject();
  return object;
});

module.exports = model("Grupo", GrupoSchema);
